smithy-cli/src/main/java/software/amazon/smithy/cli/commands/Upgrade1to2Command.java

```diff
@@ -321,6 +323,19 @@ public Void stringShape(StringShape shape) {
             return null;
         }
 
+        @Override
+        public Void listShape(ListShape shape) {
+            getDefault(shape);
+            if (shape instanceof SetShape) {
+                int lineNumber = shape.getSourceLocation().getLine();
+                writer.eraseLine(lineNumber);
+                writer.insertLine(lineNumber, "list " + shape.toShapeId().getName() + " {");
+                writer.insertLine(lineNumber, "@uniqueItems");
```

Comment:
I looked a bit closer into this as the `insertLine` at specific line with some new content seems a bit brittle, but I was able to break it using something like this (at the end of the file, unlikely but valid):

```
set SomeSet1 { member: NonBoxedInteger, }
```
the command fails with a cryptic message `Index: 35, Size: 34` (probably coming from trying to insert in an out of range index). If I have this (at the end of the file):

```
set SomeSet1 { member: NonBoxedInteger, }
set SomeSet2 { member: NonBoxedInteger, }
```
The command will output something like this 
```
@uniqueItems
list SomeSet1 {
@uniqueItems
list SomeSet2 {
```

This implementation will also remove comments like:

```
set SomeSet { // This comment is here
```
One way to solve it is just replacing the `set` part by `@uniqueItems\nlist` instead of the whole line, that will need some tweaking to the `ModelWriter` class to add the required `erase` and `insert` methods.

Reply:
Good spot!